kernel/nothread.c

```diff
@@ -14,15 +15,24 @@ bool k_is_in_isr(void)
 	return arch_is_in_isr();
 }
 
+static volatile bool k_sleep_timedout;
```

Comment:
use atomic_t for atomics

Reply:
I can use `atomic_t`, but this is not an "atomic" variable.